core/drivers/crypto/caam/include/caam_jr.h

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM Job Rings module header.
+ */
+#ifndef __CAAM_JR_H__
+#define __CAAM_JR_H__
+
+#include <caam_jr_status.h>
+
+/*
+ * Job context to enqueue/dequeue
+ */
+struct caam_jobctx {
+	uint32_t *desc;      /* reference to the descriptor */
+	uint32_t status;     /* executed job status */
+	uint32_t id;         /* Job identifier */
+	bool completion;     /* job completion flag */
+	void *context;       /* caller job context */
+	void (*callbk)(struct caam_jobctx *ctx); /* job completion callback */
+};
+
+/*
+ * Job Ring module configuration
+ */
+struct caam_jrcfg {
+	vaddr_t base;    /* CAAM virtual base address */
+	paddr_t offset;  /* Job Ring address offset */
+	int it_num;      /* Job Ring interrupt number */
+	uint8_t nb_jobs; /* Number of Jobs to managed */
+};
+
+/*
+ * Initialization of the CAAM Job Ring module
+ *
+ * @jrcfg  Job Ring Configuration
+ */
+enum CAAM_Status caam_jr_init(struct caam_jrcfg *jrcfg);
+
+/*
+ * Cancels a job id. Remove the job from SW Job array
+ *
+ * @job_id      Job Id
+ */
+void caam_jr_cancel(uint32_t job_id);
+
+/*
+ * Checks if one of the given job ids in bit mask format
+ * is complete. If none are complete, wait until timeout expired.
+ * Wait infinitively is @timeout_ms = (-1)
+ *
+ * @job_ids     Job Ids Mask
+ * @timeout_ms  Timeout in millisecond
+ */
+enum CAAM_Status caam_jr_dequeue(uint32_t job_ids, uint32_t timeout_ms);
```

Comment:
Could some other type than a fixed width type be used for `timeout_ms`, the 32-bit width seems rather arbitrary.

Reply:
Using a 8, 16 or 32 bits value will not change anything. At compilation the ARM register R1 (32 bits) will be used. Then as it's internal to the CAAM driver, the person calling this function is supposed to know what to do.

Reply:
Yes, but that's not the point, by the way there's 64-bit platforms to consider too.

With fixed width types you signal that you care about the specific properties of that type, how it overflows etc. At certain places it will stand out a bit.

You can keep it as it is if you think it's important.

Reply:
ok so is `unsigned int timeout_ms` better?

Reply:
Yes